xds/server_test.go

```diff
@@ -315,6 +315,24 @@ func (p *fakeProvider) Close() {
 	p.Distributor.Stop()
 }
 
+// setupClientOverride sets up an override for new xdsClient creation.
+func setupClientOverride(t *testing.T) func() {
```

Comment:
This code is very similar to what is being done in `setupOverrides`. Can you massage this a bit so that `setupOverrides` calls this, instead of having duplicate code?

Reply:
Good point. Done.

Reply:
Actually, I messed with it, and I think it's different enough to warrant a separate helper. The one already in the codebase sends a ref to the channel, and this one doesn't. Thoughts?

Reply:
I would be OK to ignore the returned channel in the case where you don't need it.

Reply:
I agree, but the problem is this newXDSClient is called once, and performs a blocking send. Thus, my test would block forever unless I took I took a local arg to returned channel and received. I can do this if you feel strongly, but it seems like different enough logic to warrant a separate helper for simplicity.

Reply:
Oki dokies. Feel free to leave it as is.

Reply:
Even in this test case I'm writing for race conditions, I think I want this one. Otherwise in each iteration for the for, I would need to recv from the channel to unblock the newXDSClient(), which would add a lot of code.